### DISCUSSION TITLE
Determining what a script is waiting for.

### DISCUSSION THREAD

**Discussion by PopeNobody** [nodejs/help#4906] [Q&A] [answered]
I'm curious as to whether there is, from javascript, a way to enumerate active handles and active requests in the event loop.

Specifically, I'd like to be able to find out why a node script that I think should have exited does not agree with me.

Usually, it seems, it's because I've made a promise I have not kept ... 

But it would save me some hair pulling if I could determine which of my promises are unfulfilled.

I found uv_loop_alive, and it seems to reduce to two things:  active handles and active requests ( I am assuming that req is short for request ) .. but I have no idea how to convert those back to a notion that is meaningful in javascript.

**Comment by devsnek** [comment, marked as answer]
i've used https://www.npmjs.com/package/wtfnode with great success

**Reply by PopeNobody** [reply]
Thanks so much!
